R/decisiontreeutils.R

```diff
@@ -0,0 +1,431 @@
+
+
+  the_target <- the.data[[name_y_var]]
+  if (is.numeric(the_target) && length(unique(the_target)) < 5 && !is_XDF) {
+    AlteryxMessage("The target variable is numeric, however, it has 4 or fewer unique values.", iType = 2, iPriority = 3)
+  }
+
+  if(rpart_params$cp < 0 || rpart_params$cp > 1) {
+    stop.Alteryx("The complexity parameter must be between 0 and 1. Please try again.")
+  }
+
+  if(is.na(as.numeric(config$the.cp)) && !(config$the.cp == "Auto" || config$the.cp == "")) {
+    stop.Alteryx("The complexity parameter provided is not a number. Please enter a new value and try again.")
+  }
+}
+
+
+#' Creation of components for model object evaluation
+#'
+#' @param config list of config options
+#' @param data list of datastream inputs
+#' @return list with components needed to create model
+createDTParams <- function(config, data) {
+  # use lists to hold params for rpart and rxDTree functions
+  param_list <- list()
+
+  # Determine if a compute context is being used, and whether it is XDF
+  # The mechanism for doing this depends on how the information is transfered
+  xdf_properties <- getXdfProperties("#1")
+  param_list$is_XDF <- xdf_properties$is_XDF
+  param_list$xdf_path <- xdf_properties$xdf_path
+
+  # get data param
+  the.data <- data$data_stream1
+  data_names <- names(the.data)
+  param_list$data <- quote(the.data)
+
+  # Get the field names
+  names_list <- getNamesFromOrdered(config$used.weights, data_names)
+  name_y_var <- names_list$y
+  names_x_vars <- names_list$x
+  name_weight_var <- names_list$w
+
+  # use field names to get formula param
+  param_list$dtree_formula <- makeFormula(names_x_vars, name_y_var)
+
+  # get weights param
+  param_list$weight_arg <- ifelse(config$used.weights, name_weight_var, NULL)
+  rpart_params$weights <- rxDTree_params$pweights <- weight_arg
+
+  # get method and parms params
+  if(config$select.type) {
+    if(config$classification) {
+      param_list$method <- "class"
+      if(config$use.gini) {
+        param_list$parms <- quote(list(split = "gini"))
+      } else {
+        param_list$parms <- quote(list(split = "information"))
+      }
+    } else {
+      param_list$method <- "anova"
+    }
+  }
+
+  # get surrogate param
+  if(config$use.surrogate.0) {
+    surrogate <- 0
+  } else if(config$use.surrogate.1) {
+    surrogate <- 1
+  } else {
+    surrogate <- 2
+  }
+
+  param_list$usesurrogate <- surrogate
+
+  # get max bins param
+  if(is_XDF && !is.na(as.numeric(config$max.bins))) {
+    max_bins <- config$max.bins
+    if(max_bins < 2) {
+      stop("The minimum bins is 2")
+    } else {
+      param_list$maxNumBins <- max_bins
+    }
+  }
+
+  # other parameters
+  param_list$minsplit <- config$min.split
+  param_list$minbucket <- config$min.bucket
+  param_list$xval <- config$xval.folds
+  param_list$maxdepth <- config$max.depth
+
+  param_list$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
+
```

Comment:
By the same count, I would directly modify the names in `config` so that they can be directly passed on to `rpart`. In other words, simply use the names that are used in `rpart`. This will eliminate the need to write such code, which is simply ball passing.